ONSdigital/eq-author-app
# Verify firebase tokens on the API

**DO NOT MERGE UNTIL** [https://trello.com/c/Zly4QhLC/1173-refresh-firebase-token-6](https://trello.com/c/Zly4QhLC/1173-refresh-firebase-token-6) has been done and ENV variables set in all environments

### What is the context of this PR?
JWT token is now verified against Firebase public keys.
Firebase rotates its public keys every few hours so needs to refetch if current keys are invalid.

### How to review 
1. Create eq-author-env variable to be your firebaseId. e.g. create eq-author-api/.env with `FIREBASE_PROJECT_ID="YOUR_FIREBASE_ID"`

2. login and use site as normal - ensure login and questionnaire editing work

3. Change REACT_APP_FIREBASE_PROJECT_ID & REACT_APP_FIREBASE_API_KEY env variables in eq-author to point to different firebase project and restart frontend. Open Incognito window and try to login and use site and see the "We're unable to save your progress right now." message. This step created a signed jwt token but for a different project.

4. Delete accessToken from localStorage and ensure application doesn't work
